crates/shared/src/bad_token/token_owner_finder.rs

```diff
@@ -67,6 +74,46 @@ pub struct Arguments {
     /// documentation for format details.
     #[clap(long, env)]
     pub token_owner_finder_rate_limiter: Option<RateLimitingStrategy>,
+
+    /// List of token addresses to be whitelisted as a potential token owners
+    /// For each token a list of owners is defined.
+    #[clap(
+        long,
+        env,
+        value_parser = parse_owners,
+    )]
+    pub whitelisted_owners: HashMap<H160, Vec<H160>>,
+}
+
+fn parse_owners(s: &str) -> Result<HashMap<H160, Vec<H160>>> {
+    let mut res = Default::default();
+    if s.is_empty() {
+        return Ok(res);
+    }
+    for pair_str in s.split(';') {
+        let mut split = pair_str.trim().split(':');
+        let key = split
+            .next()
+            .context("missing token address")?
+            .trim()
+            .parse()
+            .context("failed to parse address")?;
+        let values = split.next().context("missing value")?.trim();
```

Comment:
Slightly more succinctly:
```suggestion
        let (key, value) = pair_str
            .split_once(':')
            .context("missing token and owners")?;
        let key = key.trim().parse()?;
```